### PR TITLE
Add basic metrics to transflect-operator

### PR DESCRIPTION
**Pull request by juliaogris**
Add basic metrics to transflect-operator that count operations by status
(success/error), type (filter upsert/delete), ignored workloads/replicasets
popped of the queue, preprocessing errors and gauges currently managed
transflect EnvoyFilters.

Add integration tests for metrics and rework probes server to be re-usable
as metrics server. Add flags for metrics and probes ports.

In a follow up step consider implementing the workqueue.MetricsProvider in
order to observe the queue.

Clean up misc:

- Rework network address splitting
- Guard against panic when closing stopper channel
- Update Lease ID to Podname
- Change k3d cluster deletion order in Makefile
-  Add livenessProbe to sample transflect deployment

### CODE REVIEW

File: cluster-test.sh [hunk 1]
```diff
@@ -49,12 +54,27 @@ main() {
     echo 'Expecting: "routeguide-transflect" not found'
     ./out/retry-until-fail 5 5s kubectl get envoyfilter guppyecho-transflect -n guppyecho
 
+    # Check Metrics
```
Comment (jvmakine) [inline cluster-test.sh]: Maybe worth extracting as a `checkMetrics` function?

**Review by jvmakine — APPROVED** [review]
Nice

File: cmd/transflect-operator/auxservers.go [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+func newMetricsServer(addr string) *httpServer {
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+	return &httpServer{
+		Server: http.Server{
+			Addr:    addr,
+			Handler: mux,
+		},
+		name: "probes",
```
Comment (camh-) [inline cmd/transflect-operator/auxservers.go]: Is this meant to have the same name as the other server, or should this be "metrics"?

File: cmd/transflect-operator/operator.go [hunk 3]
```diff
@@ -241,6 +243,8 @@ func (o *operator) next(ctx context.Context) bool {
 	if o.shouldProcessReplicaset(rs, deployKey) {
 		if err := o.processReplicaset(ctx, rs, deployKey); err != nil {
 			log.Error().Err(err).Str("replica", rs.Name).Msg("Cannot process EnvoyFilter for Replicaset")
+		} else {
+			ignoredCounter.Inc()
```
Comment (camh-) [inline cmd/transflect-operator/operator.go]: this seems to be in the wrong location. This is incrementing the "ignored" counter when you process a replicaset and get an error. Perhaps restructure as:
```
if !o.shouldProcessReplicaset(rs, deployKey) {
    ignoredCounter.Inc()
    return true
}

if err := o.processReplicaset(ctx, rs, deployKey); err != nil {
    log.Error().Err(err).Str("replica", rs.Name).Msg("Cannot process EnvoyFilter for Replicaset")
}
return true
```

**Review by camh- — COMMENTED** [review]
looks pretty good. just a couple of concerns.

File: cluster-test.sh (hunk 1 above)
Comment (juliaogris) [reply cluster-test.sh]: ok, done.

File: cmd/transflect-operator/auxservers.go (hunk 2 above)
Comment (juliaogris) [reply cmd/transflect-operator/auxservers.go]: oups, sorry, thanks. done.

File: cmd/transflect-operator/operator.go (hunk 3 above)
Comment (juliaogris) [reply cmd/transflect-operator/operator.go]: another oups. thanks for picking this up!

Comment (juliaogris) [reply cmd/transflect-operator/operator.go]: better structure for sure.

File: cmd/transflect-operator/main.go [hunk 4]
```diff
@@ -19,6 +21,30 @@ import (
 var (
 	version   = "v0.0.0"
 	errSignal = fmt.Errorf("received shutdown signal")
+
+	// metrics
+	// filtersGauge includes filters that already exist at startup.
+	filtersGauge = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "transflect_envoyfilters",
+		Help: "The total number of EnvoyFilters deleted by transflect",
```
Comment (camh-) [inline cmd/transflect-operator/main.go]: Sorry just noticed the help message this time around. Pretty sure this is not "deleted"

**Review by camh- — APPROVED** [review]
LGTM modulo one string - sorry I didn't notice last time.

File: cmd/transflect-operator/main.go (hunk 4 above)
Comment (juliaogris) [reply cmd/transflect-operator/main.go]: don't say sorry. thank you for taking sooo much care with this. done.

**Pull request merged**